actors/miner/src/partition_state.rs

```diff
@@ -822,6 +822,43 @@ impl Partition {
         self.validate_bf_state()?;
         Ok(())
     }
+
+    // ajust epoch info of expirations_epochs and early_terminated within `Partition`
```

Comment:
This function mutates state internal, and returns a `Result<(), ...>`

Does `Self` have any invariants that may be broken?
How should a caller handle error?

E.g
```rust
/// if this function returns Err(...), the [`Partition`] is corrupt and should be discarded
```

But if that were the case, then should the method take `self`?

Reply:
This function is not expected to fail except when the `block_store` fails. In that case the transaction as a whole is expected to fail.

After this call, the partition is expected to be stored into dest deadline.